libextract/coretools.py

```diff
@@ -20,7 +20,12 @@ def proc(pairs):
     return proc
 
 
-def pipeline(data, functions):
-    for item in functions:
-        data = item(data)
-    return data
+def pipeline(data, *args):
+    try:
+        for arg in args[0:]:
+            data = arg(data)
+    except(TypeError):
```

Comment:
It was more of a proof of concept but the idea is the following:

In the case that someone want's to use ``pipeline``
They can provide either a single list (like we do with our predefined "strategies":
```python 
output = pipeline(data, STRATEGY)
```
or they can provide an indeterminate number of functions directly:

```python
output = pipeline(data, func1, func2, func3)
```